template.tpl

```diff
@@ -192,28 +252,46 @@ addEventCallback((containerId, eventData) => {
     })),
     gtm_client_name: getClientName(),
     event_data: eventDataParsed,
+    bigQueryConfig: bigQueryConfig,
+    cookies: cookies
   };
 
   if (getEventData('page_location')) {
     const parsedUrl = parseUrl(getEventData('page_location'));
     row.page_location_hostname = parsedUrl ? parsedUrl.hostname : null;
   }
 
-  const connectionInfo = {
-    projectId: data.bq_project_id,
-    datasetId: data.bq_dataset_id,
-    tableId: data.bq_table_id,
-  };
+  // insert to big query
+  if(configuration === 'big_query') {
+    BigQuery.insert(bigQueryConfig, [row], {}, () => {
+      log('BigQuery Success');
+    }, (errors) => {
+      log('BigQuery Failure');
+    });  
+    
+    data.gtmOnSuccess();
```

Comment:
The data.gtmOnSuccess() (and the one after) must be out of the `addEventCallback`.

Otherwise, this will never be called (as the addEventCallback is waiting for every tag to be succeeded before executing)